File: delightreading-server/src/main/java/com/delightreading/common/HibernateMapListUserType.java
package com.delightreading.common;

import com.fasterxml.jackson.core.type.TypeReference;

import java.util.List;
import java.util.Map;

public class HibernateMapListUserType extends JsonBasedType {
    public HibernateMapListUserType() {
        super(new TypeReference<List<Map>>() {
        });
    }

    @Override
    public Class<List<Map>> returnedClass() {
        return (Class<List<Map>>) (Object) List.class;
    }
}

File: delightreading-backend/src/main/kotlin/com/delightreading/common/HibernateMapListUserType.kt
package com.delightreading.common

import com.fasterxml.jackson.core.type.TypeReference

class HibernateMapListUserType : JsonBasedType(object : TypeReference<List<Map<*, *>>>() {
}) {

    override fun returnedClass(): Class<List<Map<*, *>>> {
        return List::class.java as Class<List<Map<*, *>>>
    }
}